src/main/java/org/gridsuite/study/server/StudyController.java

```diff
@@ -76,140 +82,144 @@
             @ApiResponse(code = 200, message = "The id of the network imported"),
             @ApiResponse(code = 409, message = "The study already exist"),
             @ApiResponse(code = 500, message = "The storage is down or a file with the same name already exists")})
-    public ResponseEntity<Void> createStudy(@PathVariable("studyName") String studyName,
-                                                  @RequestParam("caseFile") MultipartFile caseFile,
-                                                  @RequestParam("description") String description) throws IOException {
-
-        if (studyService.studyExists(studyName)) {
-            throw new ResponseStatusException(HttpStatus.CONFLICT, STUDY_ALREADY_EXISTS);
-        }
-
-        studyService.createStudy(studyName, caseFile, description);
-        return ResponseEntity.ok().build();
+    public Mono<ResponseEntity<Object>> createStudy(@PathVariable("studyName") String studyName,
+                                    @RequestParam("caseFile") MultipartFile caseFile,
+                                    @RequestParam("description") String description) throws IOException {
+        Mono<Boolean> studyExists = studyService.studyExists(studyName);
+
+        return studyExists.flatMap(b -> {
+            if (b.equals(Boolean.TRUE)) {
```

Comment:
extract to a method that asserts that the study doesn't exist: return a mono that errors when it exists, or empty when it doesn't exist ?